File: app/src/main/java/com/example/plantea/dominio/Evento.java
package com.example.plantea.dominio;

import android.app.Activity;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;

public class Evento {

    public static ArrayList<Evento> listaEventos = new ArrayList<>();
    private int id;
    private String nombre;
    private LocalDate fecha;
    private String hora;
    private String imagen;
    private int id_plan;
    private int visible;
    private int contador, identificador;
    GestionEventos gestorEventos = new GestionEventos();

    public Evento(){

    }

    public Evento(int id, String nombre, LocalDate fecha, String hora, int id_plan, String imagen) {
        this.id = id;
        this.nombre = nombre;
        this.fecha = fecha;
        this.hora = hora;
        this.id_plan = id_plan;
        this.imagen = imagen;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public String getHora() {
        return hora;
    }

    public void setHora(String hora) {
        this.hora = hora;
    }

    public int getId_plan() {
        return id_plan;
    }

    public void setId_plan(int id_plan) {
        this.id_plan = id_plan;
    }

    public String getImagen() {
        return imagen;
    }

    public void setImagen(String imagen) {
        this.imagen = imagen;
    }

    public int getVisible() {
        return visible;
    }

    public void setVisible(int visible) {
        this.visible = visible;
    }

    public ArrayList obtenerEventos(Activity actividad, LocalDate fechaSeleccionada){
        listaEventos = gestorEventos.listarEventos(actividad);
        //obtener eventos para una fecha determinada
        ArrayList<Evento> eventos = new ArrayList<>();
        for(Evento evento : listaEventos){
            if(evento.getFecha().equals(fechaSeleccionada)){
                eventos.add(evento);
            }
        }
        return eventos;
    }

    public int crearEvento(Activity actividad, Evento evento){
        identificador = gestorEventos.crearEvento(actividad, evento);
        return identificador;
    }

    public void eliminarEvento(Activity actividad, int id_evento){
        gestorEventos.eliminarEvento(actividad,id_evento);
    }

    public void cambiarVisibilidad(Activity actividad, int valor, int id_evento ) {
        gestorEventos.cambiarVisibilidad(actividad, valor, id_evento);
    }

    //Comprobar el numero de eventos visibles
    public int comprobarEventosVisible(Activity actividad) {
        contador = 0;
        contador = gestorEventos.comprobarEventosVisible(actividad);
        return contador;
    }
}

File: app/src/main/java/com/example/plantea/dominio/Evento.kt
package com.example.plantea.dominio

import android.app.Activity
import android.content.Context
import java.time.LocalDate
import java.time.LocalDateTime

class Evento {
    var id = 0
    var idUsuario: String? = null
    var nombre: String? = null
    var fecha: LocalDate? = null
    var hora: String? = null
    var visible = 0
    var cambiarVisibilidad = false
    var reminder : LocalDateTime? = null
    var idPlan = 0

    private var gestorEventos = GestionEventos()

    constructor()
    constructor(id: Int, idUsuario: String, nombre: String?, fecha: LocalDate?, hora: String?, planId: Int, cambiarVisibilidad: Boolean, reminder : LocalDateTime?) {
        this.id = id
        this.idUsuario = idUsuario
        this.nombre = nombre
        this.fecha = fecha
        this.hora = hora
        this.idPlan = planId
        this.cambiarVisibilidad = cambiarVisibilidad
        this.reminder = reminder
    }

    fun obtenerEventos(idUsuario: String, actividad: Activity?, fechaSeleccionada: LocalDate): ArrayList<Evento> {
       val listaEventos = gestorEventos.listarEventos(actividad, idUsuario)
        //obtener eventos para una fecha determinada
        val eventos = ArrayList<Evento>()
        for (evento in listaEventos) {
            if (evento.fecha == fechaSeleccionada) {
                eventos.add(evento)
            }
        }
        return eventos
    }

    fun obtenerInfoEvento(idEvento: Int, actividad: Activity?): Evento {
        return gestorEventos.obtenerInfoEvento(actividad, idEvento)
    }

    fun obtenerTodosEventos(idUsuario: String, actividad: Activity?): ArrayList<Evento> {
        return gestorEventos.listarEventos(actividad, idUsuario)
    }

    fun crearEvento(actividad: Activity?, evento: Evento): Int {
        return gestorEventos.crearEvento(actividad, evento)
    }

    fun editarEvento(actividad: Activity?, evento: Evento): Int {
        return gestorEventos.editarEvento(actividad, evento)
    }

    fun eliminarEvento(actividad: Activity?, idEvento: Int) {
        gestorEventos.eliminarEvento(actividad, idEvento)
    }

    fun cambiarVisibilidad(actividad: Context?, valor: Int, idEvento: Int) {
        gestorEventos.cambiarVisibilidad(actividad, valor, idEvento)
    }

    fun invisibiliarEvento(actividad: Context?, idEvento: Int , idUsuario: String?) {
        gestorEventos.invisibiliarEvento(actividad, idEvento, idUsuario!!)
    }

    //Comprobar el numero de eventos visibles
    fun comprobarEventoVisible(userId: String, fecha: String, actividad: Activity?): Int {
        return gestorEventos.comprobarEventoVisible(userId, fecha, actividad)
    }

    fun obtenerEventoPlan(idUsuario: String, fecha: String, context: Context?): Evento {
        return gestorEventos.obtenerTituloEvento(idUsuario, fecha, context)
    }

    fun checkEventosDia(idUsuario: String, fecha: String, context: Context?): Int {
        return gestorEventos.checkEventosDia(idUsuario, fecha, context)
    }

}